File: library/src/main/java/me/shkschneider/skeleton/helper/ContextHelper.java
package me.shkschneider.skeleton.helper;

import android.annotation.SuppressLint;
import android.app.Application;
import android.content.Context;
import android.support.annotation.NonNull;

/**
 * applicationContext()
 */
public class ContextHelper {

    @SuppressLint("StaticFieldLeak")
    private static Context CONTEXT;

    @Deprecated // Discouraged
    public static void applicationContext(@NonNull final Context context) {
        if (! (context instanceof Application)) {
            LogHelper.warning("Context is supposed to be Application's!");
        }
        CONTEXT = context;
    }

    public static Context applicationContext() {
        if (CONTEXT == null) {
            throw new NullPointerException("Context was not set!");
        }
        return CONTEXT;
    }

}

File: library/src/main/kotlin/me/shkschneider/skeleton/helper/ContextHelper.kt
package me.shkschneider.skeleton.helper

import android.annotation.SuppressLint
import android.app.Activity
import android.app.Application
import android.content.Context

/**
 * applicationContext()
 */
@SuppressLint("StaticFieldLeak")
object ContextHelper {

    private var context: Context? = null

    fun applicationContext(context: Context) {
        if (context !is Application && context !is Activity) {
            Logger.warning("Context is supposed to be Application or Activity based!")
        }
        this.context = context.applicationContext
    }

    fun applicationContext(): Context {
        context ?: run {
            throw NullPointerException("Context was not set!")
        }
        return context!!
    }

}
